### DISCUSSION TITLE
rudimentary solve for issues #4108, #10256 - "Allow dotnet watch to run external tools"

### DISCUSSION THREAD

**Discussion by Beej126** [dotnet/aspnetcore#29794] [Show and tell]
corresponds to issues: #4108, #10256 (which are locked so we can't post there)
the basic idea is monitor the dotnet watch run stdout and trigger other tool accordingly.
see powershell script here:
https://gist.github.com/Beej126/e9812b583d1e912a6a08e1af0576975f#file-dotnet_watch_monitor-ps1

example run:
```
~\source\repos\KingCounty\Kraken\KrakenStack\API [dev ≡ +3 ~6 -2 !]
> .\dotnet_watch_monitor.ps1

Press any key to abort the watcher
[01/30/2021 13:00:33] watch : Started
[01/30/2021 13:00:47] watch : Exited
[01/30/2021 13:00:47] watch : File changed: C:\Users\Beej\source\repos\KingCounty\Kraken\KrakenStack\API\KrakenStack.ServiceModel\Locators.cs
regenerating typescript definitions...
[01/30/2021 13:00:47] watch : Started
Updated: dtos.ts
Everything has been cleanly shut down. Bye =)
```

**Comment by sanguu724** [comment]
better way to achieve this is to avoid relying on parsing the dotnet watch console output and instead use a file system based approach. You can use a FileSystemWatcher to monitor project directories for changes and directly trigger external tools whenever a file is modified. This is more reliable because it reacts to actual file system events rather than terminal logs, which can change in format or timing. Once a change is detected, you can execute your required command or script (for example a build step or a TypeScript generator) using a process call. This approach is more stable, easier to maintain, and works consistently across environments compared to monitoring stdout from dotnet watch.
